aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -67,23 +111,43 @@ sub get_anaconda_version
     return $version;
 }
 
-# Returns the absolute path where the PXE file must be written.
+# Retuns the IP-based PXE file name, based on the PXE variant
+sub hexip_filename
+{
+    my ($ip, $variant) = @_;
+
+    my $hexip_str = '';
```

Comment:
do not use empty string, just `my $hexip_str;` so it returns undef in case of error